### PR TITLE
[GitHub] Documentation for GitHub connector

### PR DESCRIPTION
**Pull request by abhishekjoshi-crest**
## Closes https://github.com/elastic/connectors-python/issues/1118

This is a documentation PR for GitHub connector.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

File: docs/reference/github.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+This connector is in **beta** and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.
+
+## Usage
+
+To use this connector as a **connector client**, use the **GitHub** tile from the connectors list or **Customized connector** workflow.
+
+For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).
+
+## Get a GitHub Token
```
Comment (leemthompo) [inline docs/reference/github.md:17]:
```suggestion
## GitHub  access token
```

File: docs/reference/github.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+In order to authenticate via the GitHub, the user needs to configure GitHub Token to fetch the data from GitHub.
```
Comment (leemthompo) [inline docs/reference/github.md:19]:
```suggestion
Configure a GitHub  access token to fetch data from GitHub. 
```

File: docs/reference/github.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+To use this connector as a **connector client**, use the **GitHub** tile from the connectors list or **Customized connector** workflow.
+
+For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).
+
+## Get a GitHub Token
+
+In order to authenticate via the GitHub, the user needs to configure GitHub Token to fetch the data from GitHub. 
+
+To generate GitHub Token, user needs to follow below steps:
```
Comment (leemthompo) [inline docs/reference/github.md:21]:
```suggestion
Follow these steps to generate a GitHub  access token:
```

File: docs/reference/github.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+ - Go to GitHub Settings → Developer settings → Personal access tokens → Tokens(classic).
+ - Click `Generate new token`.
+ - Add a note and select the "repo" scope.
+ - Click `Generate token` and copy the token.
+
+## Configuration
+
+### Configure GitHub connector
+
+The following configuration fields need to be provided for setting up the connector:
```
Comment (leemthompo) [inline docs/reference/github.md:31]:
```suggestion
The following configuration fields are required:
```

File: docs/reference/github.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+#### `github_token`
+
+The GitHub Token to authenticate the GitHub instance.
```
Comment (leemthompo) [inline docs/reference/github.md:35]:
```suggestion
GitHub  access token to authenticate the GitHub instance.
```

File: docs/reference/github.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+- `*`
+
+#### `retry_count`
+
+The number of retry attempts after failed request to GitHub. Default value is `3`.
+
+### Content Extraction
+
+
+The connector uses the Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using the Apache text extraction library Tika. Supported file types eligible for extraction can be found as `TIKA_SUPPORTED_FILETYPES` in [utils.py](../connectors/utils.py) file.
```
Comment (leemthompo) [inline docs/reference/github.md:51]:
```suggestion
See [Content extraction](https://www.elastic.co/guide/en/enterprise-search/current/connectors-content-extraction.html).```

File: docs/reference/github.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+The connector syncs the following objects and entities:
+- **Repositories**
+- **Pull Requests**
+- **Issues**
+- **Files & Folder**
+
+## Sync rules
+
+- Files bigger than 10 MB won't be extracted
+- Permissions are not synced. **All documents** indexed to an Elastic deployment will be visible to **all users with access** to that Elastic Deployment.
```
Comment (leemthompo) [inline docs/reference/github.md:64]:
```suggestion
[Basic sync rules](https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html#sync-rules-basic) are identical for all connectors and are available by default.

Advanced sync rules are not available for this connector in the present version. Currently filtering is controlled by ingest pipelines.
```

File: docs/reference/github.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+The connector uses the Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using the Apache text extraction library Tika. Supported file types eligible for extraction can be found as `TIKA_SUPPORTED_FILETYPES` in [utils.py](../connectors/utils.py) file.
+
+## Documents and Sync
+
+The connector syncs the following objects and entities:
+- **Repositories**
+- **Pull Requests**
+- **Issues**
+- **Files & Folder**
+
```
Comment (leemthompo) [inline docs/reference/github.md:60]:
```suggestion
*NOTE*:
- Files bigger than 10 MB won't be extracted
- Permissions are not synced. **All documents** indexed to an Elastic deployment will be visible to **all users with access** to that Elastic Deployment

```

**Review by leemthompo — COMMENTED** [review]
Some minor tech writing cleanup, I'll defer to team for any technical points missing or requiring edits.

**Review by leemthompo — APPROVED** [review]
👍

File: docs/reference/github.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+- `*`
+
+#### `retry_count`
+
+The number of retry attempts after failed request to GitHub. Default value is `3`.
+
+### Content Extraction
+
+
+See [Content extraction](https://www.elastic.co/guide/en/enterprise-search/current/connectors-content-extraction.html).
```
Comment (seanstory) [inline docs/reference/github.md:51]: This needs more nuance, right? Since we're only fetching markdown and restructured-text files?

Comment (seanstory) [reply docs/reference/github.md:51]: It's fine if you'd rather capture that in a follow-up, as this was a recent change.

File: docs/reference/github.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+- **Files & Folder**
+*NOTE*:
+- Files bigger than 10 MB won't be extracted
+- Permissions are not synced. **All documents** indexed to an Elastic deployment will be visible to **all users with access** to that Elastic Deployment
+
+## Sync rules
+
+[Basic sync rules](https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html#sync-rules-basic) are identical for all connectors and are available by default.
+
+Advanced sync rules are not available for this connector in the present version. Currently filtering is controlled by ingest pipelines.
```
Comment (seanstory) [inline docs/reference/github.md:68]: I think advanced rules for Github were merged. https://github.com/elastic/connectors-python/pull/1324

**Review by seanstory — APPROVED** [review]

File: docs/reference/github.md (hunk 9 above)
Comment (seanstory) [reply docs/reference/github.md:51]: ah, that's going into 8.10, and this is going into 8.9. Ignore me

File: docs/reference/github.md (hunk 10 above)
Comment (seanstory) [reply docs/reference/github.md:68]: that's going into 8.10, and this is going into 8.9. Ignore me

**Pull request merged**